js/components/urlBar.js

```diff
@@ -159,6 +159,18 @@ class UrlBar extends ImmutableComponent {
     })
     // escape key handling
     ipc.on(messages.SHORTCUT_ACTIVE_FRAME_STOP, this.onActiveFrameStop.bind(this))
+    ipc.on(messages.CERT_ERROR, (e, details) => {
+      // Sometimes the cert error fires before the active frame location
+      // has been updated, so wait 100ms.
+      window.setTimeout(() => {
```

Comment:
I couldn't figure out a way to get around it otherwise. The race condition worst-case is that the page is blank (which is what happens now when you hit a page with a cert error). In this case, the user will probably reload the page and then it will definitely show about:certerror on reload.